generate/adapter_v2.py

```diff
@@ -84,10 +84,17 @@ def main(
     prompt_length = encoded.size(0)
 
     t0 = time.perf_counter()
-    y = generate(model, encoded, max_new_tokens, temperature=temperature, top_k=top_k, eos_id=tokenizer.eos_id)
+    y = generate(
+        model,
+        idx=encoded,
+        max_seq_length=max_new_tokens,
+        max_new_tokens=max_new_tokens,
+        temperature=temperature,
+        top_k=top_k,
+        eos_id=tokenizer.eos_id
+    )
     t = time.perf_counter() - t0
 
-    model.reset_cache()
```

Comment:
Why is this removed?

Reply:
good question

Reply:
Btw I didn't remove it. And it works fine without it. Is it necessary? @awaelchli do you know if this is essential for keeping memory low in case we have multiple generations?

Reply:
If the batch size (unimplemented) or `max_seq_length` changed between generations, then this would need to be called. Since none of these change right now, it could be removed. Or it might be good to keep for completeness

Reply:
I added it back, with a comment, just to avoid potential bugs in case we want to play with the `max_seq_len` some day and forget this.